irohad/torii/command_service.hpp

```diff
@@ -6,105 +6,42 @@
-    CommandService &operator=(const CommandService &) = delete;
-
-    /**
-     * Actual implementation of sync Torii in CommandService
-     * @param tx - Transaction we've received
-     */
-    void Torii(const iroha::protocol::Transaction &tx);
+    virtual ~CommandService() = default;
 
     /**
-     * Actual implementation of sync Torii in CommandService
+     * Actual implementation of sync Torii in CommandServiceImpl
      * @param tx_lis - transactions we've received
      */
-    void ListTorii(const iroha::protocol::TxList &tx_list);
-
-    /**
-     * Torii call via grpc
-     * @param context - call context (see grpc docs for details)
-     * @param request - transaction received
-     * @param response - no actual response (grpc stub for empty answer)
-     * @return - grpc::Status
-     */
-    virtual grpc::Status Torii(grpc::ServerContext *context,
-                               const iroha::protocol::Transaction *request,
-                               google::protobuf::Empty *response) override;
-
-    /**
-     * Torii call for transactions list via grpc
-     * @param context - call context (see grpc docs for details)
-     * @param request - list of transactions received
-     * @param response - no actual response (grpc stub for empty answer)
-     * @return - grpc::Status
-     */
-    virtual grpc::Status ListTorii(grpc::ServerContext *context,
-                                   const iroha::protocol::TxList *request,
-                                   google::protobuf::Empty *response) override;
+    virtual void handleTransactionList(
+        const shared_model::interface::TransactionSequence &tx_list) = 0;
+    virtual void handleTransactionListError(
+        const std::vector<shared_model::crypto::Hash> &tx_hashes,
+        const std::string &error) = 0;
 
     /**
      * Request to retrieve a status of any particular transaction
      * @param request - TxStatusRequest object which identifies transaction
      * uniquely
-     * @param response - ToriiResponse which contains a current state of
-     * requested transaction
-     */
-    void Status(const iroha::protocol::TxStatusRequest &request,
-                iroha::protocol::ToriiResponse &response);
-
-    /**
-     * Status call via grpc
-     * @param context - call context
-     * @param request - TxStatusRequest object which identifies transaction
-     * uniquely
-     * @param response - ToriiResponse which contains a current state of
+     * @param response - TransactionResponse which contains a current state of
      * requested transaction
-     * @return - grpc::Status
      */
-    virtual grpc::Status Status(
-        grpc::ServerContext *context,
-        const iroha::protocol::TxStatusRequest *request,
-        iroha::protocol::ToriiResponse *response) override;
+    virtual std::shared_ptr<shared_model::interface::TransactionResponse>
+    getStatus(const shared_model::crypto::Hash &request) = 0;
```

Comment:
Can't it be `const` method?

Reply:
Nope, there's logging at least and probably some other mutable components present

Reply:
But logging does not change state of the object, so `const`-ness is possible, if only that is the issue